zipkin-lens/src/components/DiscoverPage/DiscoverPageContent.tsx

```diff
@@ -399,29 +404,60 @@ const DiscoverPageContent: React.FC<DiscoverPageContentProps> = ({
     }
   }, [setQueryParams, tempCriteria, tempLookback, tempLimit]);
 
-  const [traces, isLoadingTraces] = useSelector((state: RootState) => [
+  const [
+    traces,
+    isLoadingTraces,
+    traceSummaries,
+  ] = useSelector((state: RootState) => [
     state.traces.traces,
     state.traces.isLoading,
+    state.traces.traceSummaries as TraceSummary[],
   ]);
 
   const [isShowingLookbackMenu, setIsShowingLookbackMenu] = useState(false);
-
   const toggleLookbackMenu = useCallback(() => {
     setIsShowingLookbackMenu((prev) => !prev);
   }, []);
-
   const closeLookbackMenu = useCallback(() => {
     setIsShowingLookbackMenu(false);
   }, []);
 
   const [isOpeningSettings, setIsOpeningSettings] = useState(false);
-
   const handleSettingsButtonClick = useCallback(() => {
     setIsOpeningSettings((prev) => !prev);
   }, []);
 
-  let content: JSX.Element | undefined;
+  const [filters, setFilters] = useState<string[]>([]);
+  const filterOptions = useMemo(
+    () =>
+      Object.keys(
+        traceSummaries.reduce((acc, cur) => {
+          cur.serviceSummaries.forEach((serviceSummary) => {
+            acc[serviceSummary.serviceName] = true;
```

Comment:
Can you use `Set` instead of object / keys?

Reply:
Just confirming are you leaving this as is or considering Set?

Reply:
Oops! I was correcting a different code.
I'll use Set here.

Reply:
Done! 🙇